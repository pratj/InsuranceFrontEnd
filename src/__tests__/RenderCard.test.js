import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import RenderCard from "../components/RenderCard";

test("renders renderCard component", () => {
  shallow(<RenderCard />);
});

test("renders renderCard component with parent div", () => {
  const wrapper = shallow(<RenderCard />);
  expect(wrapper.find('[data-test="renderCard"]').length).toBe(1);
});

// better approach is to give the button a className then test using that
test("render a button with text of `Don't know what to choose`", () => {
  const wrapper = shallow(<RenderCard />);
  expect(wrapper.find("footer").text()).toBe("Don't know what to choose?");
});

test("RenderCard snapshot", () => {
  const renderCardTree = shallow(<RenderCard />);
  expect(toJson(renderCardTree)).toMatchSnapshot();
});