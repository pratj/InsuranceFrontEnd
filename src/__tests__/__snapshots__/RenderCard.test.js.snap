// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderCard snapshot 1`] = `
<div
  className="cardRender undefined"
  data-test="renderCard"
>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={3}
  />
  <footer
    style={
      Object {
        "marginLeft": "40%",
        "marginTop": "90px",
      }
    }
  >
    <WithStyles(ForwardRef(Button))
      color="primary"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "",
          "borderRadius": "90px",
        }
      }
      variant="outlined"
    >
      <b>
        <h3>
          Don't know what to choose?
        </h3>
      </b>
    </WithStyles(ForwardRef(Button))>
  </footer>
</div>
`;
